src/domain/settings/return-reasons/detail.tsx

```diff
@@ -0,0 +1,124 @@
+import { navigate } from "gatsby"
+import {
+  useAdminDeleteReturnReason,
+  useAdminUpdateReturnReason,
+} from "medusa-react"
+import React, { useEffect } from "react"
+import { useForm } from "react-hook-form"
+import DuplicateIcon from "../../../components/fundamentals/icons/duplicate-icon"
+import TrashIcon from "../../../components/fundamentals/icons/trash-icon"
+import Input from "../../../components/molecules/input"
+import BodyCard from "../../../components/organisms/body-card"
+import DeletePrompt from "../../../components/organisms/delete-prompt"
+import useModal from "../../../hooks/use-modal"
+import useToaster from "../../../hooks/use-toaster"
+import { getErrorMessage } from "../../../utils/error-messages"
+import CreateReturnReasonModal from "./create-reason-modal"
+
+const ReturnReasonDetail = ({ reason }) => {
+  const {
+    isOpen: showDuplicateModal,
+    handleOpen: handleOpenDuplicateModal,
+    handleClose: handleCloseDuplicateModal,
+  } = useModal()
+  const {
+    isOpen: showDanger,
+    handleClose: handleClosePrompt,
+    handleOpen: handleOpenPrompt,
+  } = useModal()
+  const { register, reset, handleSubmit } = useForm()
+  const toaster = useToaster()
+  const deleteRR = useAdminDeleteReturnReason(reason?.id)
+  const updateRR = useAdminUpdateReturnReason(reason?.id)
+
+  const handleDeletion = async () => {
+    deleteRR.mutate(null)
+  }
+
+  const onSave = data => {
+    if (data.label === "") return
+    updateRR.mutate(data, {
+      onSuccess: () => {
+        toaster("Successfully updated return reason", "success")
+        navigate("/a/settings/return-reasons/")
```

Comment:
```suggestion
```

This shouldn't be necessary with the caching updates

Reply:
good catch! thanks!